lib/kubeclient.rb

```diff
@@ -43,11 +40,14 @@ def rest_client
           ssl_client_cert:  @ssl_options[:client_cert],
           ssl_client_key:   @ssl_options[:client_key]
         }
-
-        RestClient::Resource.new(@api_endpoint, options)
+        endpoint_with_version = @api_endpoint.merge(@api_endpoint.path + '/' \
+                                                    + @api_version)
```

Comment:
(Commenting on this nit only because I am already commenting below)

I prefer to leave the operator at the end of the line:

     endpoint_with_version = @api_endpoint.merge(@api_endpoint.path + '/' +
                                                 @api_version)

And I think you can spare the "\" at the end.